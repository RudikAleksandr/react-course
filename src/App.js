import React from 'react';

class ReactComponent extends React.Component {
  render() {
    return <h1>{this.props.title}</h1>;
  }
}
class ReactPureComponent extends React.PureComponent {
  render() {
    return <h1>{this.props.title}</h1>;
  }
}

const ReactFunctionComponent = (props) => {
  return <h1>{props.title}</h1>;
}

function App(props) {
  return (
    <div>
      {props.title}
      <ReactComponent title='render React Component' />
      <ReactPureComponent title='render React Pure Component' />
      <ReactFunctionComponent title='render React Function Component' />
    </div>
  );
}

export default App;

