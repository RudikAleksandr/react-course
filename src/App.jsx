import React from 'react';
import {hot} from 'react-hot-loader'

class ReactComponent extends React.Component {
  render() {
    return <h1>{this.props.title}</h1>;
  }
}
class ReactPureComponent extends React.PureComponent {
  render() {
    return <h1>{this.props.title}</h1>;
  }
}

const ReactFunctionComponent = (props) => {
  return <h1>{props.title}</h1>;
}

function App(props) {
  return (
    <div>
      {props.title}
      <ReactComponent title='render React Component' />
      <ReactPureComponent title='render React Pure Component' />
      <ReactFunctionComponent title='render React Function Component' />
    </div>
  );
}

export default hot(module)(App);

