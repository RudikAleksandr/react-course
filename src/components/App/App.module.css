.appContainer {
  background-color: #555;
}

.mainContent {
  background-color: #232323;
}

.filterSortRow {
  border-bottom: 3px solid #424242;
  box-shadow: 0 1px 0 black;
}
