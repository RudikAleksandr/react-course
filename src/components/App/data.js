import imageMovie1 from './../../assets/images/movieScreensaver/movie1.png'
import imageMovie2 from './../../assets/images/movieScreensaver/movie2.png'
import imageMovie3 from './../../assets/images/movieScreensaver/movie3.png'
import imageMovie4 from './../../assets/images/movieScreensaver/movie4.png'

const data = {
  listMovies: [
    {
      id: 1,
      photoLink: imageMovie1,
      name: 'PulpFiciton',
      type: 'Action',
      releaseYear: '2004'
    },
    {
      id: 2,
      photoLink: imageMovie3,
      name: 'PulpFiciton',
      type: 'Action',
      releaseYear: '1998'
    },
    {
      id: 3,
      photoLink: imageMovie2,
      name: 'PulpFiciton',
      type: 'Action',
      releaseYear: '2012'
    },
    {
      id: 4,
      photoLink: imageMovie3,
      name: 'PulpFiciton',
      type: 'Action',
      releaseYear: '2010'
    },
    {
      id: 5,
      photoLink: imageMovie4,
      name: 'PulpFiciton',
      type: 'Action',
      releaseYear: '2015'
    },
    {
      id: 6,
      photoLink: imageMovie1,
      name: 'PulpFiciton',
      type: 'Action',
      releaseYear: '2004'
    }
  ]
}

export default data;