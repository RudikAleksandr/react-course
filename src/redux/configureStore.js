import { configureStore as configureStoreToolkit } from '@reduxjs/toolkit';
import thunkMiddleware from 'redux-thunk';
import moviesReducer from './moviesSlice';

const configureStore = (preloadedState) => (
  configureStoreToolkit({
    preloadedState,
    reducer: {
      movies: moviesReducer,
    },
    middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(thunkMiddleware),
  })
);

export default configureStore;
