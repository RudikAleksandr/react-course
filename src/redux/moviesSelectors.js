
const selectListMovies = (state) => state.movies.moviesOrder.map(id => state.movies.movies[id]);

const selectSearch = (state) => state.movies.search;

const selectActiveMovieGenre = (state) => state.movies.activeMovieGenre;

const selectActiveSortType = (state) => state.movies.activeSortType;

export {
  selectListMovies,
  selectSearch,
  selectActiveMovieGenre,
  selectActiveSortType
};