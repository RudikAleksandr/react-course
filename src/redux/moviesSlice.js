import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import * as moviesAPI from '../api/moviesAPI';
import { groupByKey } from '../utils/grouping';

export const initialState = {
  movies: {},
  moviesOrder: [],
  search: '',
  activeMovieGenre: '',
  activeSortType: 'release_date'
};

export const getListMovies = createAsyncThunk(
  'movies/getListMovies',
  moviesAPI.getListMovies
);

export const getMovie = createAsyncThunk(
  'movies/getMovie',
  moviesAPI.getMovie
);

export const updateMovie = createAsyncThunk(
  'movies/updateMovie',
  moviesAPI.updateMovie
);

export const createMovie = createAsyncThunk(
  'movies/createMovie',
  moviesAPI.createMovie
);

export const deleteMovie = createAsyncThunk(
  'movies/deleteMovie',
  moviesAPI.deleteMovie
);

const moviesSlice = createSlice({
  name: 'movies',
  initialState,
  reducers: {
    setSearch: (state, { payload }) => {
      state.search = payload;
    },
    setActiveMovieGenre: (state, { payload }) => {
      state.activeMovieGenre = payload;
    },
    setActiveSortType: (state, { payload }) => {
      state.activeSortType = payload;
    }
  },
  extraReducers: {
    [getListMovies.fulfilled]: (state, { payload }) => {
      state.movies = groupByKey('id', payload.data);
      state.moviesOrder = payload.data.map((movie) => movie.id);
    },
    [updateMovie.fulfilled]: (state, { payload }) => {
      state.movies[payload.id] = payload;
    },
  },
});

export const {
  setSearch,
  setActiveMovieGenre,
  setActiveSortType
} = moviesSlice.actions;

export default moviesSlice.reducer;