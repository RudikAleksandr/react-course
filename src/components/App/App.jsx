import React, { Component } from 'react';
import { Container, Row, Col } from 'react-bootstrap';
import Header from '../Header';
import FilterMovies from '../FilterMovies';
import SortMovies from '../SortMovies';
import ListMovies from '../ListMovies';
import Footer from '../Footer';
import ErrorBoundary from '../ErrorBoundary';
import cn from './App.module.css';
import data from './data';
import SearchMovies from '../SearchMovies/SearchMovies';
class App extends Component {
  state = {
    listMovies: data.listMovies,
    activeMovieType: 'all',
    activeSortType: 'release date'
  };

  handleClickMovieType = (activeMovieType) => {
    this.setState({ activeMovieType });
  }

  handleSelectSortType = (activeSortType) => {
    this.setState({ activeSortType });
  }

  render() {
    return (
      <ErrorBoundary>
        <Container fluid className={cn.appContainer}>
          <Row className="flex-column align-items-center">
            <Header>
              <SearchMovies />
            </Header>
            <Col xs="12" lg="11" xl="9">
              <Row className={`flex-column align-items-center mt-2 ${cn.mainContent}`}>
                <Col className="mt-1" xs="11">
                  <Row className={cn.filterSortRow}>
                    <Col xs="8">
                      <FilterMovies
                        activeType={this.state.activeMovieType}
                        onClickMovieType={this.handleClickMovieType}
                      />
                    </Col>
                    <Col xs="4">
                      <SortMovies
                        activeType={this.state.activeSortType}
                        onSelectSortType={this.handleSelectSortType}
                      />
                    </Col>
                  </Row>
                  <ListMovies listMovies={this.state.listMovies} />
                </Col>
              </Row>
            </Col>
            <Footer />
          </Row>
        </Container>
      </ErrorBoundary>
    );
  }

};

export default App;
