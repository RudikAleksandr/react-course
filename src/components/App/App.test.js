
import { render } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router';
import App from './App';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { initialState } from '../../redux/moviesSlice';

const getComponent = (store) => (
  <MemoryRouter>
    <Route>
      <Provider store={store}>
        <App />
      </Provider>
    </Route>
  </MemoryRouter>
);

describe('App component', () => {
  test('render App snapshot', () => {
    const store = configureStore()({ movies: initialState });
    const component = getComponent(store);
    const { asFragment } = render(component);

    expect(asFragment(component)).toMatchSnapshot();
  });
});