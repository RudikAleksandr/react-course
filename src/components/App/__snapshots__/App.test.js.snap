// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component render App snapshot 1`] = `
<DocumentFragment>
  <div
    class="appContainer container-fluid"
  >
    <div
      class="flex-column align-items-center row"
    >
      <div
        class="header col"
      >
        <div
          class="justify-content-between align-items-end mb-5 ml-5 mr-5 mt-3 row"
        >
          <a
            class="netflixLink"
            href="/"
          >
            <b>
              netflix
            </b>
            roulette
          </a>
          <button
            class="addMovieButton btn"
            type="button"
          >
            + Add movie
          </button>
        </div>
        <div
          class="justify-content-center searchMovies row"
        >
          <div
            class="col-md-7 col-11"
          >
            <h1
              class="findMovieTitle"
            >
              Find your movie
            </h1>
          </div>
          <div
            class="mt-4 col-md-7 col-11"
          >
            <div
              class="justify-content-center row"
            >
              <div
                class="col-xl-8 col-8"
                xxl="9"
              >
                <div
                  class="form-group"
                >
                  <input
                    class="searchInput form-control"
                    placeholder="What do you want to watch?"
                    value=""
                  />
                </div>
              </div>
              <div
                class="pl-0 col-4"
              >
                <button
                  class="btn btn-primary"
                  type="button"
                >
                  Search
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-xl-9 col-lg-11 col-12"
      >
        <div
          class="flex-column align-items-center mt-2 mainContent row"
        >
          <div
            class="mt-1 col-11"
          >
            <div
              class="filterSortRow row"
            >
              <div
                class="col-8"
              >
                <div
                  class="filterMovies row"
                >
                  <div
                    class="pl-0 col"
                  >
                    <button
                      class="filterButton  btn"
                      type="button"
                    >
                      all
                    </button>
                    <button
                      class="filterButton  btn"
                      type="button"
                    >
                      documentary
                    </button>
                    <button
                      class="filterButton  btn"
                      type="button"
                    >
                      comedy
                    </button>
                    <button
                      class="filterButton  btn"
                      type="button"
                    >
                      horror
                    </button>
                    <button
                      class="filterButton  btn"
                      type="button"
                    >
                      crime
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="col-4"
              >
                <div
                  class="align-items-center justify-content-end sortMovies row"
                >
                  <span
                    class="sortByLabel"
                  >
                    sort by
                  </span>
                  <div
                    class="sortTypeDropdown dropdown btn-group"
                    role="group"
                  >
                    <button
                      aria-expanded="false"
                      aria-haspopup="true"
                      class="dropdown-toggle btn"
                      type="button"
                    >
                      release date
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex-column mb-4 listMoviesContainer row"
            >
              <div
                class="pl-0 mt-4 mb-4 col"
              >
                <span
                  class="countMovies"
                >
                  0
                </span>
                <span
                  class="moviesFoundLabel"
                >
                  movies found
                </span>
              </div>
              <div
                class="col"
              >
                <div
                  class="noMovieFound"
                >
                  <h2
                    class="noMovieFoundLabel"
                  >
                    No Movie Found
                  </h2>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col"
      >
        <div
          class="justify-content-center align-items-center footer row"
        >
          <a
            class="netflixLink nav-link"
            href="/home"
          >
            <b>
              netflix
            </b>
            roulette
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
